addons/l10n_in_pos/static/src/js/receipt.js

```diff
@@ -0,0 +1,28 @@
+odoo.define('l10n_in_pos.receipt', function (require) {
+"use strict";
+
+var models = require('point_of_sale.models');
+var round_pr = require('web.utils').round_precision;
+
+models.load_fields('product.product', 'l10n_in_hsn_code');
+
+var _super_orderline = models.Orderline.prototype;
+models.Orderline = models.Orderline.extend({
+    export_for_printing: function() {
+        var line = _super_orderline.export_for_printing.apply(this,arguments);
+        line.l10n_in_hsn_code = this.get_product().l10n_in_hsn_code;
+        return line;
+    },
+});
+
+var _super_order = models.Order.prototype;
+models.Order = models.Order.extend({
+    export_for_printing: function () {
+        var receipt = _super_order.export_for_printing.apply(this,arguments);
```

Comment:
In the point of sale module, you have 'client: client ? client.name : null ,' in the 'export_for_printing' function.
You can replace it by 'client: client ? client : null ,' and you won't have to do this override, and you will be able to use client.phone in the report.

P.S. check that client is not used and you should not replace it by 'client.name', but I think it is not used. You'll be able to set his name too as the specification requires it.